src/app/models/submission.server.model.ts

```diff
@@ -208,6 +213,24 @@ EncryptSubmissionSchema.statics.addWebhookResponse = function (
   ).exec()
 }
 
+EncryptSubmissionSchema.statics.retrieveWebhookInfoById = function (
+  this: IEncryptSubmissionModel,
+  submissionId: string,
+): Promise<SubmissionWebhookInfo | null> {
+  return (this.findById(submissionId)
+    .populate('form', 'webhook')
+    .exec() as Promise<IPopulatedWebhookSubmission | null>).then(
```

Comment:
nit: should we remove the `exec()` call and move the cast into the returned promise?

```ts
EncryptSubmissionSchema.statics.retrieveWebhookInfoById = function (
  this: IEncryptSubmissionModel,
  submissionId: string,
): Promise<SubmissionWebhookInfo | null> {
  return this.findById(submissionId)
    .populate('form', 'webhook')
    .then((populatedSubmission: IPopulatedWebhookSubmission | null) => {
      if (!populatedSubmission) return null
      return {
        webhookUrl: populatedSubmission.form.webhook?.url ?? '',
        isRetryEnabled: !!populatedSubmission.form.webhook?.isRetryEnabled,
        webhookView: populatedSubmission.getWebhookView(),
      }
    })
}
```